judge/runguard.c

```diff
@@ -699,15 +700,17 @@ void read_optarg_time(const char *desc, double *times)
 	/* Check for soft:hard limit separator and cut string. */
 	if ( (sep=strchr(optcopy,':'))!=NULL ) *sep = 0;
 
+	errno = 0;
 	times[0] = strtod(optcopy,&ptr);
 	if ( errno || *ptr!='\0' || !finite(times[0]) || times[0]<=0 ) {
 		error(errno,"invalid %s specified: `%s'",desc,optarg);
 	}
 
 	/* And repeat for hard limit if we found the ':' separator. */
 	if ( sep!=NULL ) {
+		errno = 0;
 		times[1] = strtod(sep+1,&ptr);
-		if ( errno || *ptr!='\0' || !finite(times[1]) || times[1]<=0 ) {
+		if ( errno || *(sep+1)=='\0' || *ptr!='\0' || !finite(times[1]) || times[1]<=0 ) {
```

Comment:
what are you checking here?

Reply:
this is checking if the number after ':' is empty

seems not necessary here because `times[1]<=0` also catches this case; but explicit check improves readability?